packages/experiments/src/utilities/BaseState.tsx

```diff
@@ -1,86 +1,57 @@
-import { IStateComponentProps } from '../Foundation';
-import { BaseComponent } from '../Utilities';
+import * as React from 'react';
 
-// TODO: add transforms and flesh out typing
-export type ITransform<TComponentProps, TViewProps> = {
-  /**
-   * Available transforms.
-   */
-  transform: 'toggle';
-  /**
-   * Name of prop that transform should change. Must exist in TViewProps.
-   */
-  prop: keyof TViewProps;
-  /**
-   * Component prop setting default value.
-   */
-  defaultValueProp?: keyof TComponentProps;
-  /**
-   * Default value for prop when defaultValueProp is undefined.
-   */
-  defaultValue: boolean;
-  /**
-   * Callback as input into transform indicating change on input. Must exist in TViewProps.
-   */
-  onInput: keyof TViewProps;
-};
-
-export type IStateTransforms<TComponentProps, TViewProps> = (ITransform<TComponentProps, TViewProps>)[];
-
-export type IBaseStateComponentProps<TComponentProps, TViewProps> = IStateComponentProps<TComponentProps, TViewProps>;
+export interface IBaseStateOptions<TState> {
+  controlledProps: (keyof TState)[];
+}
 
-// TODO: One or the other of these solutions should work in TS. TypeScript is preventing use of
-//        keyof against generic types throughout this file.
-//        https://github.com/Microsoft/TypeScript/issues/13948
-// export type IBaseStateComponentState<TViewProps> = Partial<TViewProps>;
-export type IBaseStateComponentState<TViewProps> = {
```

Comment:
Per our discussion the new changes have unlinked `TProps` and `TState` which causes type issues with passing props to state. `IBaseStateComponentProps` and `IBaseStateComponentState` were used here to bind `TComponentProps`, `TViewProps`, and component state altogether to let us move props along in a type safe way. We'll want to use some method like this I think...